packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -792,10 +811,14 @@ export class PaginatedMessage {
 
 	/**
 	 * Sets up the message.
-	 * @param channel The channel the handler is running at.
+	 *
+	 * @param messageOrInteraction The message or interaction that triggered this {@link PaginatedMessage}.
+	 * Generally this will be the command message or an interaction (either a {@link CommandInteraction} or a {@link SelectMenuInteraction}),
+	 * but it can also be another message from your client, i.e. to indicate a loading state.
+	 *
 	 * @param author The author the handler is for.
 	 */
-	protected async setUpMessage(channel: Message['channel'], targetUser: User): Promise<void> {
+	protected async setUpMessage(messageOrInteraction: Message | CommandInteraction | SelectMenuInteraction, targetUser: User): Promise<void> {
```

Comment:
Just pass down the channel and guild tbh

Reply:
The `messageOrInteraction` is also needed a few lines down so then I end up adding more parameters for no real purpose.